### PR TITLE
Line edition

### PR DESCRIPTION
**Pull request by Gamouche**
- History
- all actions keys for line editing
- syntax hightlighting is fully done

all seems to work

missing: ctrl+d, signals,  subprompts

### CODE REVIEW

**Comment by jjaniec** [comment]
From cli_options:
- Option parser & function to generate help message with passed t_option list
- Remake main to work with options
- Usages and option structs for env & cd builtins

File: srcs/line_edition/delete_char_into_cmdline_without_moving_cursor.c [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+// variables useless
+
+	actionk_cursor_move_right(le);
+	delete_char_into_cmdline_while_moving_back_cursor(le);
+
+
+
+
+
+return ;return ;return ;return ;return ;return ;return ;return ;return ;return ;return ;
```
[the remark this answers is not in the document]
Comment (jjaniec) [reply srcs/line_edition/delete_char_into_cmdline_without_moving_cursor.c]: nice

**Review by SebastienBruckert — APPROVED** [review]
Too much lines of code sorry ! :D

**Review by cgaspart — APPROVED** [review]
Approved but we still have 11 leaks on every char insert

**Review by jjaniec — APPROVED** [review]

**Pull request merged**